ray-operator/controllers/ray/common/service.go

```diff
@@ -346,15 +346,12 @@ func getServicePorts(cluster rayv1.RayCluster) map[string]int32 {
 func getPortsFromCluster(cluster rayv1.RayCluster) (map[string]int32, error) {
 	svcPorts := map[string]int32{}
 
-	if cluster.Spec.HeadGroupSpec.Template.Spec.Containers != nil &&
-		cluster.Spec.HeadGroupSpec.Template.Spec.Containers[RayContainerIndex].Ports != nil {
```

Comment:
Curious why we don't need these nil checks?

Reply:
The check is simply a workaround for easily adding tests. For more details, you can refer to the changes in #1672. We do not have the check before #1672. 

In addition, a Pod must have at least 1 container. If the PodSpec does not have any container, we should throw an exception rather than ignore it.

* [K8s doc](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#containers)
    * "List of containers belonging to the pod. Containers cannot currently be added or removed. There must be at least one container in a Pod. Cannot be updated."

Reply:
Thanks!